### DISCUSSION TITLE
MudExPropertyEdit CreateFieldForExpression forget return ？

### DISCUSSION THREAD

**Discussion by happtim** [fgilde/MudBlazor.Extensions#76] [Q&A]
The CreateFieldForExpression method in the MudExPropertyEdit class does not return anything, which means that it did not generate the <Func<TPropertyType>> for the For expression used in MudFieldText.

```
    private Expression<Func<TPropertyType>> CreateFieldForExpression<TPropertyType>()
    {
        try
        {
            Expression.Lambda<Func<TPropertyType>>(Expression.Property(Expression.Constant(PropertyMeta.ReferenceHolder, PropertyMeta.ReferenceHolder.GetType()), PropertyMeta.PropertyInfo);
        }
        catch (Exception e) { }

        return default(Expression<Func<TPropertyType>>);
    }
```

**Comment by fgilde** [comment]
Dont know where youre code is from, but it returns the correct expression otherwise whole error handling on validation will not work. The code looks like this

```c#
    private Expression<Func<TPropertyType>> CreateFieldForExpression<TPropertyType>()
    {
        try
        {
            return Expression.Lambda<Func<TPropertyType>>(Expression.Property(
                Expression.Constant(PropertyMeta.ReferenceHolder, PropertyMeta.ReferenceHolder.GetType()),
                PropertyMeta.PropertyInfo));
        }
        catch
        {
            return null;
        }
    }
```
    
See https://github.com/fgilde/MudBlazor.Extensions/blob/main/MudBlazor.Extensions/Components/ObjectEdit/MudExPropertyEdit.razor.cs#L88

**Comment by happtim** [comment]
It's possible that the latest code hasn't been pulled.

**Comment by happtim** [comment]
There is another issue. The expression generated by the CreateFieldForExpressionPropertyType() method and the anonymous lambda expression generated by the compiler have problems with the PropertyName obtained in MudBlazor.

MudBlaozr use GetFullPathOfMember get Path
```
string GetFullPathOfMember<T>( Expression<Func<T>> property)
{
	var resultingString = string.Empty;
	var p = property.Body as MemberExpression;

	while (p != null)
	{
		if (p.Expression is MemberExpression )
		{
			resultingString = p.Member.Name + (resultingString != string.Empty ? "." + resultingString : string.Empty);
		}
		p = p.Expression as MemberExpression;
	}
	return resultingString;
}
```
The path generated by CreateFieldForExpressionPropertyType() for the expression retrieval is Empty.

Below is the test code I provided.

```
var contract = new Contract();
PropertyInfo propertyInfo = typeof(Contract).GetProperty( "FirstName");
var lambda =  Expression.Lambda<Func<string>>(Expression.Property(Expression.Constant(contract, contract.GetType()), propertyInfo));

var fullPath = GetFullPathOfMember(lambda);
fullPath.ShoudleNotBe(string.Empty);

public class Contract
{
	public string FirstName { get; set; }
	public string LastName { get; set; }
	public string Email { get; set; }
}
```
![1713232569974](https://github.com/fgilde/MudBlazor.Extensions/assets/7238267/a2e273b7-66e6-4983-9d8c-476b3ca3ddb0)
